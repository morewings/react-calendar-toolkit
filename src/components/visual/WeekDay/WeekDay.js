import React from 'react';
import './WeekDay.scss';
import PropTypes from 'prop-types';

const WeekDay = props => <div className="weekday">{props.name.short}</div>;

WeekDay.propTypes = {
  name: PropTypes.shape({
    short: PropTypes.string.isRequired,
    narrow: PropTypes.string.isRequired,
    abbreviated: PropTypes.string.isRequired,
    wide: PropTypes.string.isRequired,
    numeric: PropTypes.number.isRequired,
  }).isRequired,
};

export default WeekDay;
