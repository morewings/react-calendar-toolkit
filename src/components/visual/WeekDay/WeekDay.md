`Weekday` component renders grid with weekday names on the top of `Day Calendar`. Here is how `Datepicker` with custom `Weekday` component looks like:

```js
import DatePicker from 'react-calendar-toolkit';

const style = {
  boxSizing: 'border-box',
  flex: '1 1 calc(100% / 7)',
  color: 'black',
  textAlign: 'center',
  padding: '0',
  border: '2px dashed pink',
  margin: '2px',
  fontSize: '12px',
  height: '24px',
  lineHeight: '24px'
};

const CustomComponent = ({name}) => {
  return (
    <div style={style}>
      {name.narrow}
    </div>
  )
};

<DatePicker
  onDateSet={date => {console.log('date set', date);}}
  minPrecision="day"
  renderWeekDayAs={CustomComponent}
  title="Demo datepicker" />
```

### WeekDay Component example
```js { "file": "../WeekDay.js" }
```

### Wrap WeekDay Grid with custom component
You can provide wrapper component to wrap all rendered `WeekDay`, which can be React Component or DOM node.

```js
import DatePicker from 'react-calendar-toolkit';

const style = {
  display: 'flex',
  width: 'auto',
  flexDirection: 'row',
  flexWrap: 'wrap',
  padding: '0 8px 8px',
  margin: '0 8px 8px',
  height: '24px',
  border: '2px dashed pink',
};

const CustomComponent = ({children}) => (
  <div style={style}>{children}</div>
);

<DatePicker
  onDateSet={date => {console.log('date set', date);}}
  minPrecision="day"
  wrapWeekDaysWith={CustomComponent}
  title="Demo datepicker" />
```


