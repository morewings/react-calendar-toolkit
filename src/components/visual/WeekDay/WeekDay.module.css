.grid{
  display: flex;
  border-bottom: 1px solid var(--gray);
  margin-bottom: var(--basePadding);
  padding: 0 var(--innerPadding);
  box-sizing: border-box;
}

.wrapper{
  text-align: center;
  box-sizing: border-box;
  flex: 1 1 calc(100% / 7);
  height: var(--dayHeight);
  line-height: var(--dayHeight);
  display: inline-block;
  vertical-align: middle;
  color: var(--gray);
}