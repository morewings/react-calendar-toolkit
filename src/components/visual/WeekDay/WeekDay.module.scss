@import "./../../../variables";

.grid{
  display: flex;
  border-bottom: 1px solid $gray;
  margin-bottom: $basePadding;
  padding: 0 $innerPadding;
  box-sizing: border-box;
}

.wrapper{
  text-align: center;
  box-sizing: border-box;
  flex: $dayCellFlex;
  height: $dayHeight;
  line-height: $dayHeight;
  display: inline-block;
  vertical-align: middle;
  color: $gray;
}