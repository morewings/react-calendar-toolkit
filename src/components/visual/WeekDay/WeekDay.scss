@import "./../../../variables";

.weekday{
  //margin-left: 3px;
  //width: $dayWidth;
  text-align: center;
  padding: $dayPadding;
  box-sizing: border-box;
  flex: $dayCellFlex;
}