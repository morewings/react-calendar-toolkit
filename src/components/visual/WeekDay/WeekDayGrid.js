import React from 'react';
import PropTypes from 'prop-types';

const WeekdayGrid = ({children, wrapperClassName}) => (
  <div className={wrapperClassName}>{children}</div>
);

WeekdayGrid.propTypes = {
  wrapperClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
};

WeekdayGrid.defaultProps = {
  wrapperClassName: 'weekdays-wrapper',
};

export default WeekdayGrid;
