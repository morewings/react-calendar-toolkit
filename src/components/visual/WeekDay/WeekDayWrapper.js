/*eslint-disable*/
import React from 'react';
import './WeekDayWrapper.scss'

const WeekDayWrapper = props => {
  return (
    <div className="weekdays-wrapper">{props.children}</div>
  )
};

export default WeekDayWrapper