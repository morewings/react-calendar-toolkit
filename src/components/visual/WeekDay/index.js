export {default} from './WeekDay';
export {default as WeekDayGrid} from './WeekDayGrid';
