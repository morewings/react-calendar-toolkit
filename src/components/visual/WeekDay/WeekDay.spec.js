import React from 'react';
import {DatepickerMockProvider as Provider} from 'utils/testProvider';
import {render, fireEvent} from '@testing-library/react';
import WeekDay from './WeekDay';

const defaultProps = {
  name: {
    numeric: 1,
    wide: 'wide',
    abbreviated: 'abbreviated',
    narrow: 'narrow',
  },
};

describe('WeekDay', () => {
  it(`it renders`, () => {
    const {asFragment} = render(<WeekDay {...defaultProps} />, {
      wrapper: ({children}) => <Provider>{children}</Provider>,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
