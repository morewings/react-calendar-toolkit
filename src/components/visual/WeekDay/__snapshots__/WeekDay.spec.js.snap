// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDay it renders 1`] = `
<DocumentFragment>
  <div
    aria-label="wide"
    class="wrapper"
  />
</DocumentFragment>
`;
